resolwe/observers/models.py

```diff
@@ -19,6 +19,48 @@
 Observable = PermissionObject
 
 
+class BackgroundTask(Observable):
+    """The observable model representing the background task.
+
+    When a long running task is started, the instance of type BackgroundTask is created
+    and frontend can subscribe to receive its notifications.
+    """
+
+    #: background task is preparing
+    STATUS_PREPARING = "PP"
+    #: background task is processing
+    STATUS_PROCESSING = "PR"
+    #: background task has finished successfully
+    STATUS_DONE = "OK"
+    #: background task has finished with error
+    STATUS_ERROR = "ER"
+
+    STATUS_CHOICES = (
+        (STATUS_PREPARING, "Preparing"),
```

Comment:
It would make sense to rename this to `STATUS_WAITING` as the task is waiting in queue during this stage.

Reply:
Done.